src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -241,19 +253,215 @@ uct_ib_mlx5_devx_reg_ksm_data(uct_ib_mlx5_md_t *md,
     return status;
 }
 
+static ucs_status_t
+uct_ib_mlx5_devx_reg_ksm_data_contig(uct_ib_mlx5_md_t *md, uct_ib_mlx5_mr_t *mr,
+                                     off_t off, struct mlx5dv_devx_obj **mr_p,
+                                     uint32_t *mkey)
+{
+    const size_t reg_length = UCT_IB_MD_MAX_MR_SIZE;
+    intptr_t addr           = (intptr_t)mr->super.ib->addr & ~(reg_length - 1);
+    /* FW requires indirect atomic MR addr and length to be aligned
+     * to max supported atomic argument size */
+    size_t length           = ucs_align_up(mr->super.ib->length +
+                              (intptr_t)mr->super.ib->addr - addr,
+                              md->super.dev.atomic_align);
+    int list_size           = ucs_div_round_up(length, reg_length);
```

Comment:
if `length % reg_length == 0`, `list_size == length / reg_length` and byte_count will be set to `0` on line 282.
e.g. `length == reg_length`, `list_size == 1` and in this case - `byte_count == 0`.

Reply:
nice catch, lets fix it in separate PR, this code was copy/pasted, @Artemy-Mellanox could you pls take a look

Reply:
https://github.com/openucx/ucx/pull/8071